### PR TITLE
[AMORO-3863] Add script to reformat code

### LINKED ISSUE #3863: [Improvement]: Add script to reformat code
### Search before asking

- [x] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

add script to reformat code

### How should we improve?

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by turboFei**
## Why are the changes needed?

Close #3863.

## Brief change log

- Add script to reformat code

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: dev/reformat [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# limitations under the License.
+#
+
+set -x
+
+PROJECT_HOME="$(cd "`dirname "$0"`/.."; pwd)"
+
+PROFILES="-Psupport-all-formats,spark-3.5,spark-3.3"
+
+# $PROJECT_HOME/build/mvn spotless:apply $PROFILES
```
Comment (turboFei) [inline dev/reformat]: waiting for #3849

Comment (zhoujinsong) [reply dev/reformat]: I have merged #3849. Maybe we can change it to the new one here.

Comment (turboFei) [reply dev/reformat]: done.
